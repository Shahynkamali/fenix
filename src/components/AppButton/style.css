.button {
  @apply inline-flex h-full w-full items-center text-center transition ease-in-out duration-150 font-medium rounded-md;
}
