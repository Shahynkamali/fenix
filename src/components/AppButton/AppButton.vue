<template>
  <button
    :disabled="disabled"
    class="button"
    v-on="$listeners"
  >
    <slot>
      {{ text }}
    </slot>
  </button>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import './style.css';

@Component({
  name: 'AppButton',
})

export default class AppButton extends Vue {
  @Prop({ required: false })
  readonly disabled!: boolean;

  @Prop({ required: false, default: 'click me' })
  readonly text!: string;
}
</script>
