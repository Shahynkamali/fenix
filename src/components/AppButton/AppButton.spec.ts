import { shallowMount, mount } from '@vue/test-utils';
import AppButton from './AppButton.vue';

describe('AppButton', () => {
  it('it should render the prop text', () => {
    const text = 'App';
    const wrapper = shallowMount(AppButton, {
      propsData: { text },
    });
    expect(wrapper.text()).toMatch(text);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should disable the button', () => {
    const disabled = true;
    const wrapper = mount(AppButton, {
      propsData: {
        disabled,
      },
    });
    expect(wrapper.find('button').attributes('disabled')).toBe('disabled');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Raises click event when clicked', () => {
    const onClick = jest.fn();
    const wrapper = shallowMount(AppButton, {
      listeners: {
        click: onClick,
      },
    });
    wrapper.find('button').trigger('click');
    expect(onClick).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  });
});
