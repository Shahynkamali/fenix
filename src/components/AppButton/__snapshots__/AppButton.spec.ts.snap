// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton Raises click event when clicked 1`] = `
<button
  class="button"
>
  
    click me
  
</button>
`;

exports[`AppButton it should render the prop text 1`] = `
<button
  class="button"
>
  
    App
  
</button>
`;

exports[`AppButton should disable the button 1`] = `
<button
  class="button"
  disabled="disabled"
>
  
    click me
  
</button>
`;
